### ISSUE TITLE
'fragmentshow' event only trigger on forward direction?

### ISSUE THREAD

**Issue by belync** [hakimel/reveal.js#2759]
It seems when flip slides/fragment backward, fragmentshow event not working. It only works with forward direction?

**Comment by ssokolow** [comment]
From my experimentation, it looks like `fragmentshown` is a misnomer and what's actually going on is that it would be better named `fragmentstateadvance`.

That is:
* If you have a `fade-out` shown fragment at the beginning of your stack, it doesn't fire when it appears.
* It does fire on the blank space after a `fade-out` fragment.
* `fragmenthidden` fires when fragments are being shown if you step back.

Basically, don't think of it as a "show" event. Think of it as a "cursor advanced to the next entry in the list" event.
